### ISSUE TITLE
fbcunn fails to build with "make_unique" due to lack of support in C++ 11

### ISSUE THREAD

**Issue by cshea** [facebook/fbcunn#114]
System setup - Ubuntu 16.04, gcc 5.4.0, cuda 8.0, cudnn v5, torch 7

When attempting to compile fbcunn using the supplied CMakeLists.txt and the SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11") option the code fails to build with the error "fbcunn/src/util/GlobalAsyncCopier.cpp:26:10: error: ‘make_unique’ is not a member of ‘std’
   return std::make_unique<AsyncCopier>(bufferSize << 20);" This is due to the keyword 'make_unique' not being support in c++11. 

If one switch to SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14") - the code fails to compile with the error "[  1%] Building NVCC (Device) object CMakeFiles/torch_fb_fbcunn_FFTIteratedConvolution.dir/src/fft/torch_fb_fbcunn_FFTIteratedConvolution_generated_FFTIteratedConvolution.cu.o
/usr/lib/gcc/x86_64-linux-gnu/5/include/stddef.h(436): error: identifier "nullptr" is undefined"

Please suggest a workaround.

**Comment by KevinYuk** [comment]
Hi @cshea,
Have you solved the issue?

**Comment by cshea** [comment]
No, I have not solved the issue. For now I am running in a nvidia-docker with ubuntu 14.04.

**Comment by howardzhang87** [comment]
I have solved this problem by changing `std::make_unique` in line 26 to `folly::make_unique`. Below is the explanation.

When you look at the change history of GlobalAsyncCopier.cpp, someone changed this line because he thinks folly would not support `make_unique` in the future.

**Comment by howardzhang87** [comment]
@cshea @KevinYuk 
UPDATE: You can simply change line 45 in CMakeLists.txt from
`SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")`
to
`SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++1y")`
